src/utils/workers/RenderWorker.js

```diff
@@ -1,23 +1,61 @@
 import DomMutationHandler from '../dom/DomMutationHandler.js';
 import EventForwarder from '../dom/EventForwarder.js';
+import NodeSerialiser from '../dom/NodeSerialiser.js';
 
 export default class RenderWorker extends Worker {
   constructor(scriptUrl, { container, ...options }) {
     super(scriptUrl, options);
 
-    this._domMutationHandler = new DomMutationHandler({ container });
-    this._eventForwarder = new EventForwarder((e) => this._forwardEvent(e));
+    this._nodes = new Map();
+
+    this._domMutationHandler = new DomMutationHandler({
+      container,
+      nodes: this._nodes,
+      onNodeFirstSeen: (node) => this._initialiseNode(node),
+    });
+
+    this._eventForwarder = new EventForwarder(window, (e) => this._forwardEvent(e));
 
     this.onmessage = this._handleMessage;
   }
 
+  _initialiseNode(node) {
+    this._nodes.set(node._id, node);
+
+    if (node.nodeName !== 'IFRAME') {
+      return;
+    }
+
+    // setTimeout is used here to give the frame's document a chance to be set.
+    setTimeout(() => {
+      [node, node.contentDocument].forEach((n) => {
+        n._eventForwarder = new EventForwarder(n, (e) => this._forwardEvent(e));
+      });
+    });
+  }
+
   _handleMessage({ data }) {
-    if (data.type === 'domMutations') {
-      this._domMutationHandler.processMutations(data.mutations);
+    switch (data.type) {
+      case 'domMutations':
+        this._domMutationHandler.processMutations(data.mutations);
+        break;
+
+      case 'parseHtml':
+        this._parseHtml(data);
+        break;
     }
   }
 
   _forwardEvent(event) {
     this.postMessage({ type: 'event', event });
   }
+
+  _parseHtml({ text }) {
+    const doc = new DOMParser().parseFromString(text, 'text/html');
```

Comment:
Workers don't have access to `DOMParser`, so to parse the frame's initial content string we send it to the main thread, parse it, and send the result back to the worker.